### ISSUE TITLE
Expected python (.py) files for site-packages

### ISSUE THREAD

**Issue by Luccifer** [IfcOpenShell/IfcOpenShell#272]
Hi there! 
I have completed the building and installing ifcopenshell but have only bin, include and lib folders. Any idea how to get or use it in python? Cause usually I putting compiled .py files to python/libs/site-packages folder and thats it and using it in my code as import "module" in sources

**Comment by Stinkfist0** [comment]
Running `make install` should install the Python bits to site-packages folder.

**Comment by Luccifer** [comment]
Ty! From which folder?

**Comment by Stinkfist0** [comment]
From you build directory, e.g. for me it's C:\Path\To\IfcOpenShell\build-vs2017-x84

**Comment by Luccifer** [comment]
make: *** No rule to make target 'INSTALL'.  Stop.
Lowercase as well

**Comment by thorade** [comment]
I think I had to run it from the win directory:
https://github.com/IfcOpenShell/IfcOpenShell/tree/master/win

**Comment by Luccifer** [comment]
Did from both folders

**Comment by thorade** [comment]
Yes, sorry, I did not mean make install, but the .bat file named
install-ifcopenshell.bat

**Comment by Luccifer** [comment]
I did it as well - everything build and "installed" but no any py files for site-packages

**Comment by Stinkfist0** [comment]
First of, you are using make so you're probably not on Windows, so you can ignore the Windows/MSVC scripts.

**Comment by Luccifer** [comment]
I am on windows

**Comment by Luccifer** [comment]
BuildDepsCahce(64):
```
GEN_SHORTHAND=vs2017-x64

OCC_INCLUDE_DIR=C:\Users\1\Desktop\IfcOpenShell-master\deps-vs2017-x64-installed\opencascade\inc

OCC_LIBRARY_DIR=C:\Users\1\Desktop\IfcOpenShell-master\deps-vs2017-x64-installed\opencascade\win64\lib

PYTHONHOME=C:\Python27
```

And after install-ifcopenshell.bat:
```
Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.02

x64 RelWithDebInfo IfcOpenShell installation finished.
```

**Comment by aothms** [comment]
What's the full output of `install-ifcopenshell.bat`?

**Comment by Stinkfist0** [comment]
Wonder if the IfcPython was disabled (-DIFCPYTHON=0 passed for CMake?). Could you paste more detailed output for the INSTALL project?

**Comment by aothms** [comment]
Yeah, for that matter probably also provide the contents of your `build-vs2017-x64\CMakeCache.txt` (or the bits pertaining to python).

**Comment by Luccifer** [comment]
**install output**:  https://gist.github.com/Luccifer/2f3a6a323cf7380b1e55a0d38e54a886
**CMakeCache**: https://gist.github.com/Luccifer/18f8594ae157c79ca2cb08ebeea163bd

**Comment by aothms** [comment]
@Stinkfist0 is right, in CMakeCache `BUILD_IFCPYTHON:BOOL=0`. How did that happen? Perhaps you can just edit it to `On` and rerun `install-ifcopenshell.bat`?

**Comment by Luccifer** [comment]
Ty, will try and return in a while. No idea why :(

**Comment by Luccifer** [comment]
Now the building of ifc failed with:

```
Build FAILED.

       "C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\build-vs2017-x64\ALL_BUILD.vcxproj" (default target) (1) ->
       "C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\build-vs2017-x64\ifcwrap\_ifcopenshell_wrapper.vcxproj" (defa
       ult target) (10) ->
       (CustomBuild target) ->
         CUSTOMBUILD(1): error : Unable to find 'swig.swg' [C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\build-vs2
       017-x64\ifcwrap\_ifcopenshell_wrapper.vcxproj]
         CUSTOMBUILD(3): error : Unable to find 'python.swg' [C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\build-v
       s2017-x64\ifcwrap\_ifcopenshell_wrapper.vcxproj]
         C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\src\ifcwrap\IfcPython.i(50): error : Unable to find 'std_str
       ing.i' [C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\build-vs2017-x64\ifcwrap\_ifcopenshell_wrapper.vcxproj
       ]
         C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\src\ifcwrap\IfcPython.i(51): error : Unable to find 'excepti
       on.i' [C:\Users\KarpushkinGA\Desktop\IfcOpenShell-master\build-vs2017-x64\ifcwrap\_ifcopenshell_wrapper.vcxproj]

    0 Warning(s)
    4 Error(s)

Time Elapsed 00:00:01.44

x64 RelWithDebInfo IfcOpenShell build failed
```

**Comment by aothms** [comment]
From: http://www.swig.org/Release/README

~~~
Troubleshooting
===============
In order to operate correctly, SWIG relies upon a set of library
files.  If after building SWIG, you get error messages like this,

    $ swig foo.i
    :1. Unable to find 'swig.swg'
    :3. Unable to find 'tcl8.swg'

it means that SWIG has either been incorrectly configured or
installed.  To fix this:

    1.  Make sure you remembered to do a 'make install' and that
        the installation actually worked.  Make sure you have
        write permission on the install directory.

    2.  If that doesn't work, type 'swig -swiglib' to find out
        where SWIG thinks its library is located.

    3.  If the location is not where you expect, perhaps
        you supplied a bad option to configure.  Use
        ./configure --prefix=pathname to set the SWIG install
        location.   Also, make sure you don't include a shell
        escape character such as ~ when you specify the path.

    4.  The SWIG library can be changed by setting the SWIG_LIB
        environment variable.  However, you really shouldn't
        have to do this.
~~~

Apparently something went wrong with the installation of SWIG. This should be taken care of by the build script: https://github.com/IfcOpenShell/IfcOpenShell/blob/master/win/build-deps.cmd#L398

Did you notice anything peculiar there? Perhaps redo the SWIG installation manually?

**Comment by Luccifer** [comment]
Dunno what is wrong with it... Build-deps is ok.. Everything done, but installing/building not (

**Comment by aothms** [comment]
What does `deps-vs2017-x64-installed\swigwin\swig.exe -swiglib` return?

**Comment by aothms** [comment]
Ok, that seems to be working fine then. Is there another version of SWIG installed perhaps that is somehow interfering?

And the path `C:\Users\1\Desktop\IfcOpenShell-master\deps-vs2017-x64-installed\swigwin\Lib` does contain files such as `swig.swg` and `python.swg`?

One thing you could do is just manually run the SWIG generation step and see whether that results in some new insights.

`C:\Users\1\Desktop\IfcOpenShell-master\deps-vs2017-x64-installed\swigwin\swig.exe -python -outdir C:/Users/1/Desktop/IfcOpenShell-master/build-vs2017-x64/ifcwrap -c++ -IC:/Users/1/Desktop/IfcOpenShell-master/deps-vs2017-x64-installed/oce/include/oce -IC:/Users/1/Desktop/IfcOpenShell-master/deps-vs2017-x64-installed/OpenCOLLADA/include/opencollada/COLLADABaseUtils -IC:/Users/1/Desktop/IfcOpenShell-master/deps-vs2017-x64-installed/OpenCOLLADA/include/opencollada/COLLADAStreamWriter -IC:/Users/1/Desktop/IfcOpenShell-master/deps-vs2017-x64-installed/icu/include -IC:/Users/1/Desktop/IfcOpenShell-master/deps/boost -IC:/Python27/include -IC:/Users/1/Desktop/IfcOpenShell-master/src/ifcwrap -o C:/Users/1/Desktop/IfcOpenShell-master/build-vs2017-x64/ifcwrap/IfcPythonPYTHON_wrap.cxx C:/Users/1/Desktop/IfcOpenShell-master/src/ifcwrap/IfcPython.i`

**Comment by Moult** [comment]
There is a fix for this issue described in #659 so unless the OP still has an issue, I think it is safe to close this.
